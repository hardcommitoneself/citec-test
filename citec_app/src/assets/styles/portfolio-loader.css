.common_width {
   width: 1300px;
   margin: 0rem auto;
}

/* body_area */
.PortfolioCreate .body_area {
   align-items: center;
   background: #f6f7fb;
   display: flex;
   justify-content: center;

   /* padding-bottom: 5rem; */
}

.PortfolioCreate .body_area .content_area > h1 {
   color: #6c8394;
   font-size: 30px;
   font-weight: 600;
   line-height: 45px;
   text-align: center;
}

.PortfolioCreate .boxes_wrapper {
   align-items: center;
   display: flex;
   margin-top: 3rem;
}

.PortfolioCreate .boxes_wrapper input {
   display: none;
}

.PortfolioCreate .boxes_wrapper .box {
   align-items: center;
   background: #ffffff;
   border-radius: 10px;
   cursor: pointer;
   display: flex;
   flex-direction: column;
   height: 309px;
   justify-content: center;
   margin-left: 0px;
   padding: 1.5rem;
   position: relative;
   text-align: center;
   width: 299px;
}

.PortfolioCreate .boxes_wrapper > *:last-child {
   margin-left: 2rem;
}

.PortfolioCreate .boxes_wrapper .box .checkbox_custom_circle {
   align-items: center;
   border-radius: 50%;
   border: 2px solid #c8ccd0;
   display: flex;
   height: 25px;
   justify-content: center;
   position: absolute;
   right: 20px;
   top: 20px;
   width: 25px;
}

.PortfolioCreate .boxes_wrapper .box .checkbox_custom_circle span {
   background-color: #fff;
   border-radius: 50%;
   display: block;
   height: 80%;
   width: 80%;
}

.PortfolioCreate .boxes_wrapper .box h1 {
   color: #2d405a;
   font-size: 20px;
   font-weight: 600;
   letter-spacing: 0em;
   line-height: 30px;
   margin-bottom: 0.8rem;
   margin: 1rem 0rem;
   text-align: center;
}

.PortfolioCreate .boxes_wrapper .box p {
   color: #79828d;
   font-size: 14px;
   font-weight: 400;
   letter-spacing: 0em;
   line-height: 24px;
   text-align: center;
}

.PortfolioCreate .boxes_wrapper .select {
   display: none;
}

.PortfolioCreate .boxes_wrapper input:checked + label {
   background-color: #0baca9;
}

.PortfolioCreate .boxes_wrapper input:checked + label p,
.PortfolioCreate .boxes_wrapper input:checked + label h1 {
   color: #fff;
}

.PortfolioCreate .boxes_wrapper input:checked + label .de_select {
   display: none;
}

.PortfolioCreate .boxes_wrapper input:checked + label .select {
   display: block;
}

.PortfolioCreate .boxes_wrapper input:checked + label .checkbox_custom_circle {
   border-color: #fff;
}

.PortfolioCreate footer {
   bottom: 0px;
   left: 50%;
   padding: 2rem 0rem;
   position: absolute;
   transform: translateX(-50%);
}

.PortfolioCreate footer > * {
   display: flex;
   justify-content: flex-end;
}

.PortfolioCreate footer .buttons_wrapper > * + * {
   margin-left: 1.5rem;
}

/* Portfolio Editor */

.PortfolioEditor .buttons_wrapper {
   grid-column: span 2;
   display: flex;
   justify-content: flex-end;
}

.PortfolioEditor .buttons_wrapper > * + * {
   margin-left: 1rem;
   margin-bottom: 2rem;
   font-size: 16px;
   font-weight: 400;
}

.PortfolioEditor .grid_body_area {
   display: inherit;
   grid-template-columns: 288px 1fr;
   grid-gap: 2rem;
}

.PortfolioEditor .grid_body_area aside {
   display: flex;
   flex-direction: column;
   align-items: center;
   background-color: #fff;
   border-radius: 15px;
   padding: 32px 14px;
   position: fixed;
   width: 15%;
   z-index: 1;
}

.PortfolioEditor .grid_body_area .close_icon {
   position: absolute;
   right: 20px;
   top: -10px;
   cursor: pointer;
}

.PortfolioEditor .grid_body_area.full_width {
   display: block;
}

.PortfolioEditor .grid_body_area aside > h1 {
   font-size: 26px;
   font-weight: 600;
   line-height: 39px;
   text-align: center;
   color: #2d405a;
}

.PortfolioEditor .grid_body_area aside > img {
   margin: 2rem 0rem;
}

.PortfolioEditor .grid_body_area aside div {
   border: 1.5px solid #e8eff6;
   background: #f8f9fa;
   padding: 1rem;
   display: flex;
   flex-direction: column;
   align-items: center;
   justify-content: center;
   text-align: center;
   min-height: 158px;
   width: 100%;
}

.PortfolioEditor .grid_body_area aside div h1 {
   font-size: 14px;
   font-weight: 600;
   line-height: 24px;
   letter-spacing: 0em;
   text-align: center;
   color: #0baca9;
}

.PortfolioEditor .grid_body_area aside div h1 span {
   font-size: 14px;
   font-weight: 400;
   line-height: 21px;
   letter-spacing: 0.02em;
   text-align: center;
   color: #79828d;
   display: block;
}

.PortfolioEditor .grid_body_area aside div button {
   height: 39px;
   width: 121px;
   background: #0baca9;
   color: #fff;
   border-radius: 10px;
   font-size: 13px;
   font-weight: 600;
   line-height: 22px;
   text-align: center;
   margin-top: 1rem;
}

.PortfolioEditor .grid_body_area aside > a {
   font-size: 14px;
   font-weight: 600;
   line-height: 17px;
   letter-spacing: 0.02em;
   text-align: left;
   display: flex;
   justify-content: center;
   align-items: center;
   color: #0baca9;
   margin-top: 1.5rem;
}

.PortfolioEditor .grid_body_area aside > button img {
   margin-left: 10px;
}

.PortfolioEditor .grid_body_area aside > .error_button_edit {
   font-size: 14px;
   font-weight: 400;
   line-height: 17px;
   letter-spacing: 0.02em;
   text-align: left;
   color: #fc848b;
}

.PortfolioEditor .right_side_editor {
   padding: 2rem;
   background-color: #fff;
   border-radius: 15px;
   margin-left: 20%;
}

.PortfolioEditor .right_side_editor .top_area > h1:first-child {
   font-size: 26px;
   font-weight: 700;
   line-height: 22px;
   letter-spacing: 0em;
   display: flex;
   align-items: center;
   color: #0baca9;
}

.PortfolioEditor .right_side_editor .top_area > h1:first-child img {
   margin-left: 20px;
}

.PortfolioEditor .right_side_editor .top_area > h1:last-child {
   font-size: 26px;
   font-weight: 600;
   line-height: 39px;
   letter-spacing: 0em;
   text-align: left;
   margin-top: 1rem;
}

.PortfolioEditor .grid_body_area.full_width .buttons_wrapper {
   margin-top: 2rem;
}

@media screen and (max-width: 1350px) {
   .common_width,
   .PortfolioCreate footer {
      width: 95%;
   }

   .PortfolioCreate footer > * {
      width: 100%;
   }
}

@media screen and (max-width: 1120px) {
   .PortfolioEditor .grid_body_area {
      grid-template-columns: 1fr;
   }

   .PortfolioEditor .buttons_wrapper {
      grid-column: span 1;
   }
}

@media screen and (max-width: 730px) {
   .PortfolioEditor .right_side_editor {
      overflow: auto;
   }

   .PortfolioEditor .editing_area {
      width: 700px;
   }
}

@media screen and (max-width: 650px) {
   .PortfolioCreate .body_area > * {
      width: 95%;
   }
}

@media screen and (max-width: 550px) {
   .PortfolioCreate .body_area .content_area > h1 {
      font-size: 20px;
   }

   .PortfolioCreate .boxes_wrapper .box h1 {
      font-size: 17px;
   }

   .PortfolioCreate .boxes_wrapper .box p {
      font-size: 12px;
   }

   .PortfolioCreate .body_area {
      padding-top: 18rem;
      padding-bottom: 8rem;
      overflow: auto;
   }

   .PortfolioCreate .boxes_wrapper {
      flex-direction: column;
   }

   .PortfolioCreate .boxes_wrapper > *:last-child {
      margin-left: 0px;
      margin-top: 2rem;
   }

   .PortfolioCreate footer {
      background: #f6f7fb;
   }

   .PortfolioCreate header .content_area nav > * + * {
      padding-left: 0.5rem;
   }

   .PortfolioCreate header .content_area nav li:nth-child(2) {
      padding-right: 0.5rem;
   }

   .PortfolioCreate footer {
      padding: 1rem 0rem;
   }
}

@media screen and (max-width: 500px) {
   .PortfolioEditor .right_side_editor .top_area > h1:first-child,
   .PortfolioEditor .right_side_editor .top_area > h1:last-child,
   .PortfolioEditor .grid_body_area aside > h1 {
      font-size: 20px;
      line-height: unset;
   }
}

.scroll::-webkit-scrollbar {
   width: 9px;
}

.scroll::-webkit-scrollbar-track {
   background: transparent;
}

.scroll::-webkit-scrollbar-thumb {
   background-color: #b0b0b0;
   border-radius: 10px;
   border: 3px solid #ffffff;
}
