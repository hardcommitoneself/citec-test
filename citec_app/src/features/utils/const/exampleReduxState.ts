// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.

export const exampleReduxState = {
   errors: '',
   taxHarvesting: {
      arrayPositionSelected: [
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      positionsTable: [
         {
            outstanding_balance: 50000,
            ticker: 'AAPL',
            unrealized_capital_gains: -1000,
            taxAlpha: 400,
         },
         {
            outstanding_balance: 0,
            ticker: '$CASH',
            unrealized_capital_gains: 500,
            taxAlpha: -200,
         },
      ],
      taxAlpha: -400,
      unrealizedCapitalGains: -500,
      unrealizedCapitalGainsNegative: -1000,
      unrealizedCapitalGainsSelected: 0,
   },
   portfolio: {
      ticker: ['$CASH', 'AAPL'],
      outstanding_balance: [1000000, 50000],
      unrealized_capital_gains: [0, 11234],
      analysis: {
         fundamentals: {
            total_assets: 2,
            total_positions: 2,
            pe: 24.32,
            pb: 38.96,
            dividend_yield: 0.59,
         },
         risk: {
            sharpe: 0.305683,
            volatility: 0.0147,
            beta: 0.048719,
            alpha: 0.000035,
         },
         returns: {
            returns_1M: -0.0021,
            returns_3M: -0.0037,
            returns_6M: -0.0028,
            CAGR: -0.1058,
         },
         capital_gains: {
            outstanding_balance: 1050000,
            unrealized_cap_losses: 0,
            unrealized_cap_gains: 11234,
            realized_cap_losses: 0,
            realized_cap_gains: 0,
            taxes_cost: 0,
            taxes_rate: 0.4,
         },
         sectors: {
            sector: ['Cash', 'Information Technology'],
            percentage: [0.952381, 0.047619],
            percentage_benchmark: [0.3333333333333333, 0.6666666666666666],
         },
         factors: {
            factor: ['style', 'size'],
            percentage: [0.502991, 0.52381],
         },
         segment_matrix: {
            style: ['Val', 'Val'],
            size: ['Large', 'Mid'],
            weight: [0, 0],
         },
         top_holdings: {
            ticker: ['$CASH', 'AAPL'],
            weight: [0.952381, 0.047619],
            outstanding_balance: [1000000, 50000],
            unrealized_capital_gains: [0, 11234],
         },
         concentration: {
            ticker: ['$CASH', 'AAPL'],
            concentration: [95.238095, 100],
         },
         var: {
            quantile: [2, 4],
            value: [0.000072, 0.001334],
         },
         risk_returns: {
            ticker: ['portfolio', 'benchmark'],
            returns: [0.004722, 0.119508],
            volatility: [0.008561, 0.205196],
            sharpe: [0.551568, 0.582407],
         },
      },
   },
   universe: {
      factors_filter: [
         {
            factor: 'style',
            ranges: [0, 100],
         },
         {
            factor: 'size',
            ranges: [0, 100],
         },
         {
            factor: 'yield',
            ranges: [0, 100],
         },
      ],
      sectors_filter: [
         'Industrials',
         'Financials',
         'Health Care',
         'Consumer Staples',
         'Communication Services',
         'Information Technology',
         'Consumer Discretionary',
         'Materials',
         'Cash',
         'Real Estate',
         'Energy',
         'Utilities',
      ],
      to_exclude: [],
      to_include: [],
      universe_base: 'SPDR S&P 500',
   },
   rebalance: {
      ticker: ['$CASH', 'AAPL'],
      hold: [0, 1],
      objective: {
         name: 'Minimize tracking error',
         key: 'min_tracking_error',
      },
      benchmark: 'SPY',
      min_weight: 0.000001,
      number_stocks: [10, 200],
   },
   portfolio_optimized: {
      portfolio_id: null,
      portfolio: 'optimized_portfolio',
      benchmark: 'SPY',
      creation_date: null,
      ticker: ['AAPL', 'MSFT'],
      outstanding_balance: [72506.52, 63477.37],
      unrealized_capital_gains: [0, 0],
      optimization_status: 'Optimization terminated successfully',
   },
   trades: [
      {
         ticker: 'MRK',
         operation: 'SELL',
         volume: 275161.76,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 3053,
         expected_volume: 275193,
         delta_volume: 31,
         price: 90.1388,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/2eedcaa6-80ec-11ea-853f-00000000092a.png',
            name: 'Merck & Co Inc',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'BMY',
         operation: 'SELL',
         volume: 152166.26,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 2072,
         expected_volume: 152146,
         delta_volume: -19,
         price: 73.43,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/8b628318-80eb-11ea-93a2-00000000092a.png',
            name: 'Bristol-Myers Squibb Co',
         },
         gain_loss: [0, 0],
      },
      {
         ticker: 'WMT',
         operation: 'SELL',
         volume: 83898.27,
         realized_capital_gains: 0,
         unrealized_capital_gains: 0,
         stocks_number: 679,
         expected_volume: 83845,
         delta_volume: -53,
         price: 123.4832,
         columnName: {
            url_logo:
               'https://static.finnhub.io/logo/52057505dd05a752d364065fcddf7772a700b31c47c39e9dcb41e393f6693d52.png',
            name: 'Walmart Inc',
         },
         gain_loss: [0, 0],
      },
   ],
   workflow: {
      type: 'load',
   },
   static_data: {
      universe: {
         universes: [
            {
               name: 'INVESCO QQQ TRUST SERIES 1',
               description: 'INVESCO QQQ TRUST SERIES 1',
            },
            {
               name: 'ISHARES MSCI USA ESG SELECT',
               description: 'ISHARES MSCI USA ESG SELECT',
            },
         ],
         sectors: [
            'Cash',
            'Communication Services',
            'Consumer Discretionary',
            'Consumer Staples',
         ],
         factors: ['style', 'size', 'yield', 'quality'],
      },
      rebalance: {
         objectives: [
            {
               key: 'min_tracking_error',
               name: 'Minimize tracking error',
            },
            {
               key: 'min_risk_parity',
               name: 'Minimize risk parity',
            },
            { key: 'min_volatility', name: 'Minimize volatility' },
            {
               key: 'max_diversification_ratio',
               name: 'Maximize diversification ratio',
            },
         ],
         benchmarks: [
            {
               key: 'sp_500',
               description: 'S&P 500 universe',
               ticker: 'SPY',
            },
            {
               key: 'nasdaq_100',
               description: 'NASDAQ 100',
               ticker: 'QQQ',
            },
         ],
      },
   },
   brokers: {
      brokerConnected: '',
      balance: 0,
      accountType: [],
      a1c_t: '',
   },
   ideas: [
      {
         portfolio_id: 16,
         portfolio: 'my_first_portfolio_sadasd',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 18,
         portfolio: 'testing',
         benchmark: 'SPY',
         creation_date: '2022-04-06 16:10:19',
         ticker: ['MSFT', 'AAPL'],
         outstanding_balance: [200000, 100000],
         unrealized_capital_gains: [0, 35700],
         optimization_status: null,
      },
      {
         portfolio_id: 19,
         portfolio: 'Jueves',
         benchmark: 'SPY',
         creation_date: '2022-06-23 10:38:00',
         ticker: [
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'AAPL',
            'MMM',
            'TMO',
            'BLK',
            'MSFT',
            'QCOM',
            'NVDA',
            'CSX',
            'ADBE',
            'XOM',
            'AVGO',
            'BAC',
            'TSLA',
            'SHW',
            'TXN',
            'NFLX',
            'GOOGL',
            'C',
            'MMC',
            'BKNG',
            'COST',
            'NKE',
            'DIS',
            'EL',
            'CSCO',
            'BRK.B',
            'META',
            'AMZN',
            'DG',
            'CCI',
            'ORCL',
            'USB',
            'TGT',
            'PG',
            'ICE',
            'GOOG',
            'AMD',
            'MCD',
            'CVS',
            'HD',
            'CME',
            'MU',
            'UNH',
            'ATVI',
            'V',
            'EOG',
            'GS',
            'HUM',
            'COP',
         ],
         outstanding_balance: [
            80352.37, 74072.31, 59343.2, 54736.44, 41924.05, 41510.68, 35441.43,
            27349.74, 26308.48, 26136.99, 25646.73, 25479.44, 24726.53, 24075.27,
            23856.18, 23763.38, 23735.12, 23092.02, 22995.95, 21888.81, 21775.5,
            21090.31, 16734.94, 16192.87, 15720.33, 14020.04, 14018.61, 13225.5,
            12996.74, 11693.52, 11315.64, 10798.53, 10583.72, 10309.54, 10245.54,
            9611.94, 8747.21, 8723.18, 6339.86, 5866.02, 5426.94, 4912.02, 4857.23,
            3915.03, 3395.13, 3370.41, 3322.34, 3184.94, 3001.78, 1983.75, 1858.2,
            1789.12, 1309.98, 1228.49,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 20,
         portfolio: 'Min volatility',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:25:41',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
      {
         portfolio_id: 21,
         portfolio: 'min2',
         benchmark: 'SPY',
         creation_date: '2022-06-23 11:27:32',
         ticker: [
            'MRK',
            'BMY',
            'WMT',
            'CMCSA',
            'REGN',
            'VTR',
            'RTX',
            'NEE',
            'MMM',
            'CSX',
            'XOM',
            'SHW',
            'TXN',
            'C',
            'BRK.B',
            'BA',
            'BSX',
            'CB',
            'CI',
            'CL',
            'CME',
            'COP',
            'COST',
            'CRM',
            'ZTS',
            'ADBE',
            'AMZN',
            'AVGO',
         ],
         outstanding_balance: [
            275161.76, 152166.26, 83898.27, 80352.37, 74072.31, 59343.2, 54736.44,
            41924.05, 35441.43, 24726.53, 23856.18, 22995.95, 21888.81, 16734.94,
            11315.64, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49, 2116.49,
            2116.49, 2116.49, 2116.49, 119.48, 96.79, 4.71,
         ],
         unrealized_capital_gains: [
            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
            0, 0, 0, 0,
         ],
         optimization_status: null,
      },
   ],
}
