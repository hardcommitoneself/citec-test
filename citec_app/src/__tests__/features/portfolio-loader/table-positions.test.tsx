// Copyright (C) 2021-Present CITEC Inc. <https://citecsolutions.com/>
// All rights reserved
//
// This file is part of CITEC Inc. source code.
// This software framework contains the confidential and proprietary information
// of CITEC Inc., its affiliates, and its licensors. Your use of these
// materials is governed by the terms of the Agreement between your organisation
// and CITEC Inc., and any unauthorised use is forbidden. Except as otherwise
// stated in the Agreement, this software framework is for your internal use
// only and may only be shared outside your organisation with the prior written
// permission of CITEC Inc.
// CITEC Inc. source code can not be copied and/or distributed without the express
// permission of CITEC Inc.
import '@testing-library/jest-dom'
import { render, fireEvent } from '@testing-library/react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { exampleReduxState } from 'features/utils/const/exampleReduxState'
import { TablePositions } from 'features/portfolio-loader'

const excelData = [
   {
      outstanding_balance: 111,
      ticker: 'AAPL',
      unrealized_capital_gains: 5000,
   },
]

describe('test at table positios component', () => {
   const mockStore = configureStore()
   const store = mockStore(exampleReduxState)

   const setExcelData = jest.fn()

   beforeEach(() => {
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      Element.prototype.scrollTo = () => {}
   })

   test('Snapshot', () => {
      const { container } = render(
         <Provider store={store}>
            <TablePositions excelData={excelData} setExcelData={setExcelData} />
         </Provider>
      )

      expect(container).toMatchSnapshot()
   })

   test('render table and simulate delete data', () => {
      const { getByTestId } = render(
         <Provider store={store}>
            <TablePositions excelData={excelData} setExcelData={setExcelData} />
         </Provider>
      )

      fireEvent.click(getByTestId('btn-delete'))

      expect(setExcelData).toHaveBeenCalledTimes(1)
      expect(getByTestId('table-positions')).toBeTruthy()
      expect(getByTestId('btn-delete')).toBeTruthy()
   })
})
