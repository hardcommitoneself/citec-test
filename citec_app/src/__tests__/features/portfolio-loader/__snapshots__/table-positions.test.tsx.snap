// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test at table positios component Snapshot 1`] = `
<div>
  <div
    class="mb-2"
    data-testid="table-positions"
  >
    <div
      class="shadow-sm rounded-lg  my-4"
    >
      <table
        class=" w-full"
        data-testid="table-test"
        role="table"
        style="min-width: 120px;"
      >
        <thead
          class="bg-[#F5F6F8]"
        >
          <tr
            role="row"
            style="display: flex; flex: 1 0 auto; min-width: 120px;"
          >
            <th
              class="p-3 text-neutral-500 font-[500]"
              colspan="1"
              role="columnheader"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px; cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex"
              >
                Ticker | Company name 
                <div
                  class=" text-neutral-400 flex flex-col mx-2"
                >
                  <div
                    class="flex flex-col justify-center items-center"
                  >
                    <div
                      class=" false caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                    <div
                      class=" rotate-180 caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="p-3 text-neutral-500 font-[500]"
              colspan="1"
              role="columnheader"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px; cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex"
              >
                Investment, $
                <div
                  class=" text-neutral-400 flex flex-col mx-2"
                >
                  <div
                    class="flex flex-col justify-center items-center"
                  >
                    <div
                      class=" false caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                    <div
                      class=" rotate-180 caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="p-3 text-neutral-500 font-[500]"
              colspan="1"
              role="columnheader"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px; cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex"
              >
                Capital gain, $
                <div
                  class=" text-neutral-400 flex flex-col mx-2"
                >
                  <div
                    class="flex flex-col justify-center items-center"
                  >
                    <div
                      class=" false caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                    <div
                      class=" rotate-180 caret"
                    >
                      <svg
                        fill="none"
                        height="10"
                        viewBox="0 0 8 4"
                        width="10"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.15102 0.113281L7.7459 3.8492H0.556131L4.15102 0.113281Z"
                          fill="#C4C4C4"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </th>
            <th
              class="p-3 text-neutral-500 font-[500]"
              colspan="1"
              role="columnheader"
              style="box-sizing: border-box; flex: 50 0 auto; min-width: 30px; width: 50px; cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="flex"
              >
                
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            class="bg-[#CBEBEB] children-table"
            role="row"
          >
            <td>
              <form
                autocomplete="off"
                class="flex flex-1 form-table relative"
              >
                <div
                  class="input-table  relative"
                >
                  <input
                    class=" input-common   input-valid"
                    data-testid="ticker"
                    name="ticker"
                    placeholder="Ticker | Company Name"
                    required=""
                    type="text"
                    value=""
                  />
                  <div
                    class="absolute top-[62px] w-full bg-white custom_cell py-2 max-h-80 rounded-lg shadow-md selector overflow-y-scroll z-20 scroll"
                    style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden; transform-origin: 0 0 0;"
                  >
                    <p
                      class="text-neutral-400 text-center"
                      data-testid="type-ticker"
                    >
                      Please type the ticker
                    </p>
                    <div
                      class="flex flex-col items-center justify-center"
                    />
                  </div>
                </div>
                <input
                  class="input-table input-common input-valid"
                  id="outstanding_balance"
                  inputmode="numeric"
                  min="1"
                  placeholder="$"
                  required=""
                  type="text"
                  value=""
                />
                <input
                  class="input-table input-common  input-valid"
                  id="unrealized_capital_gains"
                  inputmode="numeric"
                  min="1"
                  placeholder="$"
                  required=""
                  type="text"
                  value=""
                />
                <div
                  class="delete-row"
                >
                  <button
                    class="btn-add-row bg-green-600 hover:bg-green-700 "
                    data-testid="btn-add-row"
                    disabled=""
                    type="submit"
                  >
                    +
                  </button>
                </div>
              </form>
            </td>
          </tr>
          <tr
            class=""
            role="row"
            style="display: flex; flex: 1 0 auto; min-width: 120px;"
          >
            <td
              class="border-b border-neutral-300/70 p-3 text-neutral-500 "
              role="cell"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px;"
            >
              <input
                class="input-common mx-1"
                readonly=""
                value="AAPL"
              />
            </td>
            <td
              class="border-b border-neutral-300/70 p-3 text-neutral-500 "
              role="cell"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px;"
            >
              <input
                class="input-common mx-1"
                readonly=""
                value="$ 111"
              />
            </td>
            <td
              class="border-b border-neutral-300/70 p-3 text-neutral-500 "
              role="cell"
              style="box-sizing: border-box; flex: 150 0 auto; min-width: 30px; width: 150px;"
            >
              <input
                class="input-common mx-1"
                readonly=""
                value="$ 5,000"
              />
            </td>
            <td
              class="border-b border-neutral-300/70 p-3 text-neutral-500 "
              role="cell"
              style="box-sizing: border-box; flex: 50 0 auto; min-width: 30px; width: 50px;"
            >
              <button
                class="btn-delete mt-2"
                data-testid="btn-delete"
              >
                -
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
